src/main/java/net/sf/jabref/gui/maintable/MainTableNameFormatter.java

```diff
@@ -0,0 +1,40 @@
+package net.sf.jabref.gui.maintable;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.model.entry.AuthorList;
+
+public class MainTableNameFormatter {
+
+    /**
+     * Format a name field for the table, according to user preferences.
+     *
+     * @param nameToFormat The contents of the name field.
+     * @return The formatted name field.
+     */
+    public static String formatName(String nameToFormat) {
+        // Read name format options:
+        boolean namesNatbib = Globals.prefs.getBoolean(JabRefPreferences.NAMES_NATBIB); //MK:
+        boolean namesLastOnly = Globals.prefs.getBoolean(JabRefPreferences.NAMES_LAST_ONLY);
+        boolean namesAsIs = Globals.prefs.getBoolean(JabRefPreferences.NAMES_AS_IS);
+        boolean abbr_names = Globals.prefs.getBoolean(JabRefPreferences.ABBR_AUTHOR_NAMES); //MK:
+        boolean namesFf = Globals.prefs.getBoolean(JabRefPreferences.NAMES_FIRST_LAST);
+        boolean namesLf = !(namesAsIs || namesFf || namesNatbib || namesLastOnly); // None of the above.
+
+        if (nameToFormat == null) {
+            return null;
```

Comment:
I would put this right at the beginning of the method. So that the rest of the code does not have think about null at all.